src/components/WalletStatementModal/index.js

```diff
@@ -1,24 +1,51 @@
 import React from 'react';
 import {withOnyx} from 'react-native-onyx';
 import lodashGet from 'lodash/get';
+import {ActivityIndicator, View} from 'react-native';
 import compose from '../../libs/compose';
 import withLocalize from '../withLocalize';
 import ONYXKEYS from '../../ONYXKEYS';
 import {walletStatementPropTypes, walletStatementDefaultProps} from './WalletStatementModalPropTypes';
+import styles from '../../styles/styles';
+import themeColors from '../../styles/themes/default';
 
-const WalletStatementModal = (props) => {
-    const authToken = lodashGet(props, 'session.authToken', null);
-    return (
-        <iframe
-            src={`${props.statementPageURL}&authToken=${authToken}`}
-            title="Statements"
-            height="100%"
-            width="100%"
-            seamless="seamless"
-            frameBorder="0"
-        />
-    );
-};
+class WalletStatementModal extends React.Component {
+    constructor(props) {
+        super(props);
+
+        this.state = {
+            isLoading: true,
+        };
+    }
+
+    render() {
+        const authToken = lodashGet(this.props, 'session.authToken', null);
+        return (
+            <>
+                {this.state.isLoading && (
+                    <View style={[styles.flex1, styles.alignItemsCenter, styles.justifyContentCenter]}>
+                        <ActivityIndicator
+                            color={themeColors.spinner}
+                            size="large"
+                            style={[styles.walletStatementModalLoadingIndicator]}
```

Comment:
We have a component called `FullScreenLoadingIndicator` that would make it so you don't have to include these styles

Reply:
Ah, was thinking that would cover the whole screen but it does exactly what I need it to do. Will update!